src/sentry/tasks/statistical_detectors.py

```diff
@@ -335,73 +336,144 @@ def query_transactions_timeseries(
+            [
+                BooleanCondition(
+                    BooleanOp.AND,
+                    [
+                        Condition(Column("project_id"), Op.EQ, project_id),
+                        Condition(Column("transaction"), Op.EQ, transaction_name),
+                    ],
                 )
-                yield project_id, transaction_name, formatted_result
+                for project_id, transaction_name in transactions
+            ],
+        )
+        query = Query(
+            match=Entity(EntityKey.GenericMetricsDistributions.value),
+            select=[
+                Column("project_id"),
+                Function(
+                    "arrayElement",
+                    (
+                        CurriedFunction(
+                            "quantilesIf",
+                            [0.95],
+                            (
+                                Column("value"),
+                                Function("equals", (Column("metric_id"), duration_metric_id)),
+                            ),
+                        ),
+                        1,
+                    ),
+                    "p95_transaction_duration",
+                ),
+                Function(
+                    "transform",
+                    (
+                        Column(f"tags_raw[{transaction_name_metric_id}]"),
+                        Function("array", ("",)),
+                        Function("array", ("<< unparameterized >>",)),
+                    ),
+                    "transaction",
+                ),
+            ],
+            groupby=[
+                Column("transaction"),
+                Column("project_id"),
+                Function(
+                    "toStartOfInterval",
+                    (Column("timestamp"), Function("toIntervalSecond", (3600,)), "Universal"),
+                    "time",
+                ),
+            ],
+            where=[
+                Condition(Column("org_id"), Op.IN, list(org_ids)),
+                Condition(Column("project_id"), Op.IN, list(project_ids)),
+                Condition(Column("timestamp"), Op.GTE, start),
+                Condition(Column("timestamp"), Op.LT, end),
+                Condition(Column("metric_id"), Op.EQ, duration_metric_id),
+                transactions_condition,
+            ],
+            orderby=[
+                OrderBy(Column("project_id"), Direction.ASC),
+                OrderBy(Column("transaction"), Direction.DESC),
+                OrderBy(
+                    Function(
+                        "toStartOfInterval",
+                        (Column("timestamp"), Function("toIntervalSecond", (3600,)), "Universal"),
+                        "time",
+                    ),
+                    Direction.ASC,
+                ),
+            ],
+            granularity=Granularity(interval),
+            limit=Limit(10000),
+        )
+        request = Request(
+            dataset=Dataset.Events.value,
```

Comment:
maybe I'm missing something but shouldn't this be the metrics dataset?

Reply:
I'll leave this for @volokluev but fwiw this works as is, as the top transactions query uses it too. The `Query` includes `match=Entity(EntityKey.GenericMetricsDistributions.value),` which might be taking care of this

Reply:
oh weird, yeah the dataset is wrong but also datasets are kind of useless 🙊 in snuba. We should change it to not be confusing though

Reply:
Good to know 😅 Anyway, updated!